### PR TITLE
KAFKA-12747: Fix flakiness in shouldReturnUUIDsWithStringPrefix

### PR DESCRIPTION
**Pull request by guozhangwang**
Consecutive UUID generation could result in same prefix.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
cc @ableegoldman for reviews.

**Review by ableegoldman — APPROVED** [review]
LGTM, thanks for the quick triage and patch

**Comment by jlprat** [comment]
You can also take a look at https://github.com/apache/kafka/pull/10662/files#diff-944ff4a8c4eab2003ab210f5a2fceac1d124e71fcfe383b736add11803a4fd4cR468 (in there I was checking that in case of prefix clash, the content was either "a" or "b"

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java [hunk 1]
```diff
@@ -451,17 +453,14 @@ public void shouldReturnUUIDsWithStringPrefix() {
...
 
         while (keysWithPrefix.hasNext()) {
-            final KeyValue<Bytes, byte[]> next = keysWithPrefix.next();
-            valuesWithPrefix.add(new String(next.value));
+            keysWithPrefix.next();
             numberOfKeysReturned++;
         }
 
-        assertThat(numberOfKeysReturned, is(1));
-        assertThat(valuesWithPrefix.get(0), is("a"));
```
Comment (jlprat) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:464]: I would still try to keep this assertion, something among the lines of:
```java
if (clashOnPrefix) {
    assertThat(valuesWithPrefix.get(0), either(is("a")).or(is("b")));
} else {
    assertThat(valuesWithPrefix.get(0), is("a"));
}
```

Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:464]: SG, I will do that upon merging.

**Comment by guozhangwang** [comment]
Will cherry-pick to old branches as well.

**Pull request merged**